File: recursive_addition/ruby_recursive_addition/recursive_addition.rb
#!/usr/bin/ruby -w
#comments

def rec_add(x, y)
  if y == 0 then
    return x
  else
    return (1 + rec_add(x, y - 1))
  end
end

def main()
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  puts "rec_add(6,8)= #{rec_add(6,8)}"
end

main()

File: recursive_addition/python_recursive_addition/recursive_addition.py
import sys
#import sys, re, os

def rec_add(x, y):
  if y == 0:
    return x
  else:
    return (1 + rec_add(x, y - 1))

def main():
  if len(sys.argv) != 1:
    print ("Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  print (rec_add(5, 10))
  sys.exit(), end

main()
